tt_metal/jit_build/build.cpp

```diff
@@ -147,6 +147,7 @@ JitBuildState::JitBuildState(const JitBuildEnv& env, const JitBuiltStateConfig &
 void JitBuildState::finish_init() {
     if (this->is_fw_) {
         this->defines_ += "-DFW_BUILD ";
+        this->defines_ += "-DNOC_MODE=" + std::to_string(DM_DEDICATED_NOC) + " ";
```

Comment:
Is this needed? I thought adding `callback("NOC_MODE", std::to_string(NOC_MODE::DM_DEDICATED_NOC));`  will set this define?

Reply:
the thing is, callback only compiles the define for Kernels, not FW. and FW (brisc/erisc) requires this NOC_MODE for it to properly link with noc_non_blocking_api.h

Reply:
after addressing Paul's change of noc_non_blocking_api.h, this defien can be removed